src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -0,0 +1,1517 @@
+static bool gdb_client_connected = false;     ///< True if client connected
+static bool gdb_server_is_running = false;    ///< True if server is running
+static bool new_gdb_client_connected = false; ///< True if new client connected
+
+static SOCKET gdb_server_socket = SOCK_ERR_INVALID; ///< The main gdb server socket
+static SOCKET gdb_client_socket = SOCK_ERR_INVALID; ///< The gdb client socket
+
+#define UART_DEBUG_INPUT_BUFFER_SIZE 32
+static unsigned char local_uart_debug_buffer[UART_DEBUG_INPUT_BUFFER_SIZE] = {
+	0}; ///< The local buffer[ input buffer size]
+
+static SOCKET uart_debug_server_socket = SOCK_ERR_INVALID;
+static SOCKET uart_debug_client_socket = SOCK_ERR_INVALID;
+static bool uart_debug_client_connected = false;
+static bool user_configured_uart = false;
+static bool uart_debug_server_is_running = false;
+static bool new_uart_debug_client_conncted = false;
+
+static SOCKET swo_trace_server_socket = SOCK_ERR_INVALID;
+static SOCKET swo_trace_client_socket = SOCK_ERR_INVALID;
+static bool swo_trace_client_connected = false;
+static bool swo_trace_server_is_running = false;
+static bool new_swo_trace_client_conncted = false;
+
+#define SWO_TRACE_INPUT_BUFFER_SIZE 32
+static unsigned char local_swo_trace_buffer[SWO_TRACE_INPUT_BUFFER_SIZE] = {
+	0}; ///< The local buffer[ input buffer size]
+
+#define WPS_LOCAL_TIMEOUT 30 // Timeout value in seconds
+
+//
+// Sign-on message for new UART data clients
+//
+static char uart_client_signon[] = "\r\nctxLink UART connection.\r\nPlease enter the UART setup as baud, bits, parity, "
+								   "stop.\r\ne.g. 38400,8,N,1\r\n\r\n";
+
+typedef enum wi_fi_app_states {
+	app_state_wait_for_driver_init,         ///< 0
+	app_state_read_mac_address,             ///< 1
+	app_state_connect_to_wifi,              ///< 2
+	app_state_wait_wifi_disconnect_for_wps, ///< 3
+	app_state_wait_wifi_disconnect_for_http,
+	app_state_connect_wps,               ///< 5
+	app_state_wait_wps_event,            ///< 6
+	app_state_http_provision,            ///< 7
+	app_state_wait_provision_event,      ///< 8
+	app_state_wait_for_wifi_connect,     ///< 9
+	app_state_start_server,              ///< 10
+	app_state_wait_for_server,           ///< 11
+	app_state_error,                     ///< 12
+	app_state_check_default_connections, ///< 13
+	app_state_spin                       ///< 14
+} app_states_e;
+
+app_states_e app_state; ///< State of the application
+/*
+ * Define the send queue, this is used in the socket event callback
+ * to correctly process output and sync with ACKs
+ */
+#define SEND_QUEUE_SIZE        4
+#define SEND_QUEUE_BUFFER_SIZE 1024
+
+#define BUTTON_PRESS_WPS               2500 // Enter WPS mode if 2.5 Seconds > 2.5S < 5 Seconds
+#define BUTTON_PRESS_HTTP_PROVISIONING 5000
+#define BUTTON_PRESS_MODE_CANCEL       7500
+
+typedef struct {
+	unsigned char packet[SEND_QUEUE_BUFFER_SIZE]; ///< The packet[ send queue buffer size]
+	unsigned len;                                 ///< The length
```

Comment:
I made a apass through this file and the associated header file:
unsigned char -> uint8_t
unsigned -> uint32_t
unsigned in -> uint32_t

Since this change had major effects when done using rebase, I chose to add a new commit.